Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -86,11 +102,89 @@ endRideTransaction driverId bookingId ride mbFareParams mbRiderDetailsId merchan
     QRB.updateStatus bookingId SRB.COMPLETED
     DriverStats.updateIdleTime driverId
     DriverStats.incrementTotalRidesAndTotalDist (cast ride.driverId) (fromMaybe 0 ride.chargeableDistance)
+    mbDailyRideStats <- QDailyStats.findByIdAndDate ride.driverId rideDate
```

Comment:
these queries can be outside transaction this transaction and in a `fork` thread (also you can use runNoTransaction blindly for the queries where only one update or insert is going on).